src/projection/fit.js

```diff
@@ -0,0 +1,48 @@
+import {default as geoStream} from "../stream";
+import boundsStream from "../path/bounds";
+
+function fit(project, extent, object) {
+
+  var w = extent[1][0] - extent[0][0],
+      h = extent[1][1] - extent[0][1],
+      precision = project.precision(),
+      clip = project.clipExtent && project.clipExtent();
+
+  project.scale(1)
+      .translate([0, 0]);
+
+  if (clip !== undefined) {
+    project.clipExtent(null);
+  }
+
+  geoStream(object, project.stream(boundsStream));
+
+  var b = boundsStream.result(),
+      s = 1 / Math.max((b[1][0] - b[0][0]) / w, (b[1][1] - b[0][1]) / h),
+      x = extent[0][0] + (w - s * (b[1][0] + b[0][0])) / 2,
+      y = extent[0][1] + (h - s * (b[1][1] + b[0][1])) / 2;
```

Comment:
You’ll need to coerce `+extent[0][1]` here since a string value would result in unexpected behavior. However above when you subtract it’s not required (unless you want to be really paranoid and protect against non-deterministic valueOf).